app/routes/donorschoosebot.js

```diff
@@ -0,0 +1,182 @@
+'use strict';
+
+/**
+ * Imports.
+ */
+const express = require('express');
+const router = express.Router(); // eslint-disable-line new-cap
+const donorschoose = require('../../lib/donorschoose');
+const helpers = require('../../lib/helpers');
+const logger = app.locals.logger;
+
+/**
+ * Setup.
+ */
+const COUNT_FIELD = process.env.DONORSCHOOSE_DONATION_COUNT_FIELDNAME || 'ss2016_donation_count';
+const MAX_DONATIONS_ALLOWED = (process.env.DONORSCHOOSE_MAX_DONATIONS_ALLOWED || 5);
+
+/**
+ * Sends response object and posts update to user's Mobile Commons Profile to send SMS message.
+ * @param {object} res - Express response
+ * @param {number} code - Response HTTP code to send
+ * @param {string} msgType - The type of DonorsChooseBot message to send
+ * @param {object} profileUpdate - field values to update on current User's Mobile Commons Profile
+ */
+function sendResponse(res, code, msgType, profileUpdate) {
+  logger.debug(`dcbot sendResponse:${msgType} profileUpdate:${JSON.stringify(profileUpdate)}`);
+
+  const property = `msg_${msgType}`;
+  const responseMessage = app.locals.donorsChooseBot[property];
+  // TODO: Post to Mobile Commons Profile
+
+  return helpers.sendResponse(res, 200, responseMessage);
+}
+
+router.post('/', (req, res) => {
+  let incomingMessage = req.body.args;
+  if (!incomingMessage) {
+    return helpers.sendResponse(res, 422, 'Missing required args.');
+  }
+
+  incomingMessage = helpers.getFirstWord(incomingMessage);
+  logger.debug(`dcbot incomingMessage:${incomingMessage}`);
+
+  const profileFieldName = `profile_${COUNT_FIELD}`;
+  const numDonations = req.body[profileFieldName];
+
+  if (numDonations >= MAX_DONATIONS_ALLOWED) {
+    return sendResponse(res, 200, 'max_donations_reached');
+  }
+
+  // If start param passed, initiate conversation, ignoring incomingMessage.
+  const prompt = req.query.start;
+
+  if (!req.body.profile_postal_code) {
+    if (prompt) {
+      return sendResponse(res, 200, 'ask_zip');
```

Comment:
It's not meant to be an error -- Mobile Commons posts an entire Profile in `req.body` -- this  check is to see if there's a value stored in that field. If not, we send a message to User asking for zip code. Send a 200 code back to Mobile Commons mData saying Yup I know how to respond to this, I sent a `profile_update` request.